File: Mobile Programming/Mobile Programming/app/src/main/java/com/example/uasmobileprogramming/ApiInterface.java
package com.example.uasmobileprogramming;

import android.renderscript.Sampler;

import com.example.uasmobileprogramming.model.Data;
import com.example.uasmobileprogramming.model.Item;

import retrofit2.Call;
import retrofit2.http.DELETE;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.PUT;
import retrofit2.http.Path;

public interface ApiInterface {
    @GET("item")
    Call<Item> getData();
    @FormUrlEncoded
    @POST("item")
    Call<Item> tambahItem(@Field("nama") String nama,
                          @Field("harga") Integer harga,
                          @Field("jumlah") Integer jumlah);
    @FormUrlEncoded
    @PUT("item/{id}")
    Call<Item> ubahItem(@Path("id") Integer id,
                        @Field("nama") String nama,
                        @Field("harga") Integer harga,
                        @Field("jumlah") Integer jumlah
                        );

    @DELETE("item/{id}")
    Call<Item> hapusItem(@Path("id") Integer id);
}

File: app/src/main/java/com/example/belajar2/ApiInterface.kt
package com.example.belajar2

import com.example.belajar2.Model.Item
import retrofit2.Call
import retrofit2.http.DELETE
import retrofit2.http.Field
import retrofit2.http.FormUrlEncoded
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.PUT
import retrofit2.http.Path

interface ApiInterface {
    @get:GET("item")
    val data: Call<Item>

    @FormUrlEncoded
    @POST("item")
    fun tambahItem(
        @Field("nama") nama: String,
        @Field("harga") harga: Int?,
        @Field("jumlah") jumlah: Int?
    ): Call<Item>

    @FormUrlEncoded
    @PUT("item/{id}")
    fun ubahItem(
        @Path("id") id: Int?,
        @Field("nama") nama: String,
        @Field("harga") harga: Int?,
        @Field("jumlah") jumlah: Int?
    ): Call<Item>

    @DELETE("item/{id}")
    fun hapusItem(@Path("id") id: Int?): Call<Item>
}